torch/testing/_internal/composite_compliance.py

```diff
@@ -106,114 +109,138 @@ def is_inplace(func):
     return name[-1] == '_'
 
 
-class CompositeCompliantTensor(torch.Tensor):
-    elem: torch.Tensor
-
-    __slots__ = ['elem']
-
-    @staticmethod
-    def __new__(cls, elem, *args, **kwargs):
-        # The storage of CompositeCompliantTensor should never be used directly
-        # by a Composite operation; if the Composite
-        # operator attempts to read from the storage without dispatching then it'll
-        # raise a RuntimeError due to it being a meta storage.
-        r = torch.Tensor._make_wrapper_subclass(  # type: ignore[attr-defined]
-            cls, elem.size(),
-            dtype=elem.dtype, layout=elem.layout,
-            device=elem.device, requires_grad=elem.requires_grad,
-            strides=elem.stride(), storage_offset=elem.storage_offset())
-
-        # CompositeCompliantTensor steals the "requires_grad"-ness.
-        if elem.requires_grad:
-            # Why clone? Because sometimes OpInfo shares inputs between tests...
-            r.elem = elem.detach().clone()
-        else:
-            r.elem = elem
-
-        # Propagate conjugate bits to the wrapper tensor
-        # Ref: https://github.com/albanD/subclass_zoo/issues/24
-        # Ref: https://github.com/albanD/subclass_zoo/issues/21
-        torch._C._set_conj(r, r.elem.is_conj())
-        torch._C._set_neg(r, r.elem.is_neg())
-        return r
-
-    def __repr__(self):
-        return f"CompositeCompliantTensor({self.elem})"
-
-    @classmethod
-    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
-        def unwrap(e):
-            return e.elem if isinstance(e, CompositeCompliantTensor) else e
-
-        def wrap(e):
-            return CompositeCompliantTensor(e) if isinstance(e, torch.Tensor) else e
-
-        if func.overloadpacket.__name__ in ('set_', 'resize_'):
-            raise RuntimeError(
-                f"{func.__name__} is not allowed to be called inside of "
-                f"Composite operators.")
-
-        if is_inplace(func):
-            # NB: We are making an assumption that if the function is in-place,
-            # then the first argument is being written to. Introspection please save us!
-            mutated_argument = args[0]
-            if not isinstance(mutated_argument, CompositeCompliantTensor) and \
-                    any([isinstance(a, CompositeCompliantTensor) for a in args[1:]]):
+def generate_cct(dispatch=False, alias=True):
```

Comment:
nit: let's name these something a little more descriptive

dispatch -> "prevent_infinite_loops_with_modes", alias -> "workaround_autograd_view_assert_in_debug_mode"

Reply:
(feel free to try to shorten the names, I just typed the first things that came to mind)

Reply:
Going with `enable_torch_dispatch` and `autograd_view_consistency`, does that sound ok?

Reply:
enable_torch_dispatch is a bit misleading because we are using torch_dispatch in the test

Reply:
How about `enable_recursive_torch_dispatch`?

Reply:
sgtm! Thanks for going through the bikeshedding with me